cli/setup-owl.sh

```diff
@@ -11,34 +11,74 @@ rack_tag="v9.0"
 sadl_image="sadl/sadl-eclipse"
 sadl_tag="v3.5.0-20211204"
 
-mode="copy"
+mode="docker"
 
 usage() {
-        echo "Usage: setup-owl.sh [-h] [-b] [-n NAME] [-t TAG]"
+        echo "Usage: setup-owl.sh [-h] [-b] [-n NAME] [-t TAG] [-i IPADDRESS]"
         echo "   -h    print help"
         echo "   -b    build OWL files using sadl-eclipse"
         echo "   -n    specify rack-box docker image name"
         echo "   -t    specify rack-box tag when copying OWL files from running image"
+        echo "   -i    get OWL files from VirtualBox VM by IP address"
         exit 1
 }
 
-while getopts "bn:t:" o; do
+while getopts "bi:n:t:" o; do
         case "$o"
         in
                 b) mode="build";;
                 n) rack_image="${OPTARG}";;
                 t) rack_tag="${OPTARG}";;
+                i) virtualbox_ip="${OPTARG}"; mode="virtualbox";;
                 *) usage;;
         esac
 done
 
 case "${mode}"
 in
     build)
+        echo "[setup-owl] Building OWL files using docker"
+
         docker run --rm -u 0 -e RUN_AS="$(id -u) $(id -g)" -v "${rack_dir}:/RACK" ${sadl_image}:${sadl_tag} -importAll /RACK -cleanBuild
         ;;
 
-    copy)
+    virtualbox)
+        echo "[setup-owl] Copying CDR and OWL files from VirtualBox"
+
+        if ! ssh -i rack_ssh_key -oBatchMode=true "ubuntu@${virtualbox_ip}" true
+        then
+                echo "[setup-owl] Unable to authenticate to VM, attempting to install key"
+
+                if [ ! -e rack_ssh_key ]
+                then
+                        echo "[setup-owl] Generating a fresh SSH keypair"
+                        ssh-keygen -q -N "" -f rack_ssh_key
+                fi
+
+                echo "[setup-owl] Please confirm the SSH host key if prompted."
+                echo "[setup-owl] Please enter the default password if prompted: $(tput bold)ubuntu$(tput sgr0)"
+                ssh "ubuntu@${virtualbox_ip}" "mkdir -p ~/.ssh/; cat >> ~/.ssh/authorized_keys" < rack_ssh_key.pub
+        fi
+
+        echo "[setup-owl] Copying OwlModels"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/RACK-Ontology/OwlModels" "${rack_dir}/RACK-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/GE-Ontology/OwlModels" "${rack_dir}/GE-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/GrammaTech-Ontology/OwlModels" "${rack_dir}/GrammaTech-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/STR-Ontology/OwlModels" "${rack_dir}/STR-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/Boeing-Ontology/OwlModels" "${rack_dir}/Boeing-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/LM-Ontology/OwlModels" "${rack_dir}/LM-Ontology/"
+        scp -q -i rack_ssh_key -r "ubuntu@${virtualbox_ip}:RACK/SRI-Ontology/OwlModels" "${rack_dir}/SRI-Ontology/"
```

Comment:
As a thought (perhaps for the future), it would be great if we could `for dir in $(ssh ubuntu@${virtualbox_ip} find RACK -type d -name OwlModels) ; do scp ...; done`